File: task-manager-backend/src/main/java/com/taskmanager/dto/DashboardSummary.java
package com.taskmanager.dto;

import lombok.*;

import java.util.List;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class DashboardSummary {

    private long totalTasks;
    private long pendingTasks;
    private long completedTasks;
    private long todayTasks;

    private List<ChartItemDTO> statusChart;
    private List<ChartItemDTO> weeklyChart;
    private List<TaskResponse> recentTasks;

}

File: task-manager-frontend/src/app/models/dashboard-summary.ts
import { ChartItem } from "./chart-item";
import { RecentTask } from "./recent-task";

export interface DashboardSummary {

  totalTasks: number;

  pendingTasks: number;

  completedTasks: number;

  todayTasks: number;

  statusChart: ChartItem[];

  weeklyChart: ChartItem[];

  recentTasks: RecentTask[];

}